src/main/java/com/tw/go/plugin/BuildStatusNotifierPlugin.java

```diff
@@ -102,6 +106,9 @@ private GoPluginApiResponse handleGetPluginSettingsConfiguration() {
         response.put(PLUGIN_SETTINGS_USERNAME, createField("Username", null, true, false, "2"));
         response.put(PLUGIN_SETTINGS_PASSWORD, createField("Password", null, true, true, "3"));
         response.put(PLUGIN_SETTINGS_OAUTH_TOKEN, createField("OAuth Token", null, true, true, "4"));
+        if (provider.pluginId().equals(GerritProvider.PLUGIN_ID)) {
```

Comment:
The `GerritProvider` class won't be available inside other plugins, causing a CNFE here. Do you feel it's better to push this behaviour into the `Provider` interface, the provider could be passed the response object so it can add more fields into it, and determine the correct index it needs to start from, based on the size of the existing fields.